File: PO_py/program/Organisms/Animals/Wolf.py
from program.Organisms.Animal import Animal


class Wolf(Animal):
    Wolf_INITIATIVE = 5
    Wolf_STRENGTH = 9

    def __init__(self, world, x, y):
        super().__init__(world, x, y, self.Wolf_INITIATIVE, self.Wolf_STRENGTH, 'O')
        print("Wolf created")

    def get_name(self):
        return "Wolf"

    def create_child(self, x, y):
        return Wolf(self.world, x, y)

File: PO_java/program/Organisms/Animals/Wolf.java
package projekt.program.Organisms.Animals;

import projekt.program.Organism;
import projekt.program.Organisms.Animal;
import projekt.program.World;

import java.util.Random;

public class Wolf extends Animal {

    private static final int WOLF_INITIATIVE = 5; // Example value, set according to your game's rules
    private static final int WOLF_STRENGTH = 9;  // Example value, set according to your game's rules

    public Wolf(World world, int x, int y) {
        super(world, x, y, WOLF_INITIATIVE, WOLF_STRENGTH, 'W');
        System.out.println("Wolf created");
    }

    @Override
    public String getName() {
        return "Wolf";
    }

    @Override
    protected Organism createChild(int x, int y) {
        return new Wolf(this.world, x, y);
    }
}

